qiskit/quantum_info/synthesis/xx_decompose/polytopes.py

```diff
@@ -0,0 +1,266 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Defines bare dataclasses which house polytope information, as well as a specialized data structure
+which describes those two-qubit programs accessible to a given sequence of XX-type interactions.
+"""
+
+from copy import copy
+from dataclasses import dataclass, field
+from itertools import combinations
+import random
+from typing import List
+
+import numpy as np
+
+from qiskit.exceptions import QiskitError
+
+from .utilities import EPSILON
+
+
+@dataclass
+class ConvexPolytopeData:
+    """
+    The raw data underlying a ConvexPolytope.  Describes a single convex
+    polytope, specified by families of `inequalities` and `equalities`, each
+    entry of which respectively corresponds to
+
+        inequalities[j][0] + sum_i inequalities[j][i] * xi >= 0
+
+    and
+
+        equalities[j][0] + sum_i equalities[j][i] * xi == 0.
+    """
+
+    inequalities: List[List[int]]
+    equalities: List[List[int]] = field(default_factory=list)
+    name: str = ""
+
+
+@dataclass
+class PolytopeData:
+    """
+    The raw data of a union of convex polytopes.
+    """
+
+    convex_subpolytopes: List[ConvexPolytopeData]
+
+
+def polytope_has_element(polytope, point):
+    """
+    Tests whether `polytope` contains `point.
+    """
+    return all(
+        -EPSILON <= inequality[0] + sum(x * y for x, y in zip(point, inequality[1:]))
+        for inequality in polytope.inequalities
+    ) and all(
+        abs(equality[0] + sum(x * y for x, y in zip(point, equality[1:]))) <= EPSILON
+        for equality in polytope.equalities
+    )
+
+
+def manual_get_vertex(polytope, seed=42):
+    """
+    Returns a single random vertex from `polytope`.
+    """
+    random.seed(seed)
```

Comment:
I think this sets a global seed which is undesirable, see: https://github.com/Qiskit/qiskit-terra/issues/2850

Instead let's use `default_rng()` like here: 
https://github.com/Qiskit/qiskit-terra/blob/db3b35e4a3ad27457aba5525f5f7a025c32973b9/qiskit/quantum_info/states/random.py#L39

Reply:
Cool, I didn't know local seeds were possible in python. There is a bunch of seed-setting in the test suite too; I'll do the same there.